.travis.yml

```diff
@@ -50,13 +57,29 @@ script:
         mkdir -p "${HOME}/.foldercache/${exercise}/.stack-work"
         ln -f -s "${HOME}/.foldercache/${exercise}/.stack-work"
 
-        # Here we prepare the exercise to be tested by Stack.
-        MODULE=`sed -n 's/ *module \+\([a-zA-Z0-9]\+\).*/\1/p' src/Example.hs`
-        mv src/Example.hs "src/${MODULE}.hs"
+        if [ -f src/Example.hs ]; then
+            # Here we prepare the exercise to be tested by Stack.
+            MODULE=`sed -n 's/ *module \+\([a-zA-Z0-9]\+\).*/\1/p' src/Example.hs`
+            mv src/Example.hs "src/${MODULE}.hs"
+
+            test_exercise
+        else
+            found=
+            for example in examples/*; do
+                echo "testing ${example}"
+                rm -f src/*.hs
+                mv ${example}/*.hs src
+                mv ${example}/package.yaml .
+                found=1
+
+                test_exercise
+            done
+
+            if [ -z "$found" ]; then
```

Comment:
I think this does not work as intended. On my computer, at least, if `examples` dir is empty or does not exist, this is wha thappens:

```
testing examples/*/
mv: cannot stat 'examples/*//*.hs': No such file or directory
mv: cannot stat 'examples/*//package.yaml': No such file or directory
```

As you can see, since the glob found nothing, it is interpreted literally. And then `mv` fails, which I assume will abort the script as it should. But that means this `found` is unnecessary.

So, is ther ea point in keeping it?